arrays/arrays.c

```diff
@@ -118,11 +168,34 @@ void arr_remove(Array *arr, char *element) {
 
   // Search for the first occurence of the element and remove it.
   // Don't forget to free its memory!
+  // Set an index
+  int ind = 0;
+
+  for (int i = 0; i < arr->count; i++) {
+    ind = i;
 
-  // Shift over every element after the removed element to the left one position
+    if (arr->elements[i] == element) {
 
-  // Decrement count by 1
+      // Set to not have an invalid pointer
+      arr->elements[i] = NULL;
+      free(arr->elements[i]);
 
+      // Shift over every element after the removed element to the left one position
+      for (int i = ind; i < arr->count; i++) {
```

Comment:
This is a good first pass solution but there are other implementations that don't require nested for loops. You could consider utilizing a counter or using a nested conditional that shifts and  free elements under different circumstances. There are a few different options here.